graylog2-server/src/main/java/org/graylog/plugins/views/search/validation/QueryValidationServiceImpl.java

```diff
@@ -66,6 +69,14 @@ public ValidationResponse validate(ValidationRequest req) {
             decoratedQuery(req);
         } catch (SearchException searchException) {
             return ValidationResponse.error(toExplanation(searchException));
+        } catch (MissingEnterpriseLicenseException licenseException) {
```

Comment:
I've seen that you are testing this change with integration test in #3303 PR of [graylog-plugin-enterprise] project.
Don't you believe that it would be nicer to try to write some unit tests for this class, inside this project?

Reply:
Yes, it will require a lot of mocking, especially if not only your change was to be tested, but the whole class, which has no unit tests at all. It is a slight indication that this class probably does too much (does not have one responsibility) and should be split. It is probably not something that needs to be part of this bug fix, but if you believe it is worth an effort, a new task may be created to clean this code a little bit.

Reply:
From my POV, the whole QueryValidationService is just a converter and transforms data from the query parser and query decorators. It doesn't contain much logic on its own. From that reason, I rather unit-tested the query parser itself and integration-tested the whole service, together with the QueryValidationResource. I'll create a task and assign it to you, this will spread the validation knowledge between both of us. Thanks!